junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertTimeout.java

```diff
@@ -180,4 +182,30 @@ static void assertTimeoutPreemptively(Duration timeout, Executable executable, S
 		}
 	}
 
+	/**
+	 * The thread factory used for preemptive timeout.
+	 *
+	 * This factory is the same with {@link Executors#defaultThreadFactory()} but provides more meaningful thread names,
+	 * helpful for debugging purposes.
+	 *
+	 */
+	private static class TimeoutThreadFactory implements ThreadFactory {
+		private static final AtomicInteger threadNumber = new AtomicInteger(1);
+		private final ThreadGroup group;
+
+		TimeoutThreadFactory() {
+			SecurityManager s = System.getSecurityManager();
+			group = (s != null) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
```

Comment:
There's no need to do this, `Thread` already does this if no group is specified explicitly.

Reply:
I think it makes sense if the factory is about to create many threads but given that at the moment we have a factory per thread it can be removed.